src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -647,27 +651,34 @@ def handleEnforceGUIDInFileName(self):
         if self.crabPSet:
             return
 
-        # only check this if we are in the first step of a workflow as output files of chained processing
-        # steps will not have correct GUID information
-        if self.step.data._internal_name != "cmsRun1":
-            self.logger.info("Not evaluating enforceGUIDInFileName parameter for step %s",
-                             self.step.data._internal_name)
-            return
+        if secondaryInput:
+            inputSource = secondaryInput
+            self.logger.info("Evaluating enforceGUIDInFileName parameter for secondary input data.")
+        else:
+            inputSource = self.process.source
 
         # only enable if source is PoolSource or EmbeddedRootSource
-        if self.process.source.type_() not in ["PoolSource", "EmbeddedRootSource"]:
+        if inputSource.type_() not in ["PoolSource", "EmbeddedRootSource"]:
             self.logger.info("Not evaluating enforceGUIDInFileName parameter for process source %s",
-                             self.process.source.type_())
+                             inputSource.type_())
             return
 
         self.logger.info("Evaluating if release %s supports enforceGUIDInFileName parameter...",
                          self.getCmsswVersion())
 
         # enable if release supports enforceGUIDInFileName and parameter is not set
         if (isEnforceGUIDInFileNameSupported(self.getCmsswVersion()) and
-                not hasattr(self.process.source, "enforceGUIDInFileName")):
+                not hasattr(inputSource, "enforceGUIDInFileName")):
+            # check to make sure primary input files follow guid naming convention
+            # prevents enabling guid checks on some workflows (StoreResults/StepChain) that use custom input file names
+            # EmbeddedRootSource input files will always follow guid naming convention
+            if inputSource.type_() == "PoolSource" and inputSource.fileNames:
+                guidRegEx = re.compile("[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}.root$")
```

Comment:
I was going to suggest to move it to a global level in this module (such that we only compile it once), but it really makes no different. This module is executed only once per cmsRun anyways

Reply:
Since this only runs once for a single file, I think any change I made would just be for readability. Just tried to rework this a couple ways, but nothing looked any better.